### PR TITLE
fix: text area input does not focus when active and add highlight input is always focussed in post a highlight form

### LINKED ISSUE #2715: Bug: Text area and Input style inconsistency in "Post a highlight" form
### Describe the bug

The textarea for adding information on the highlight form does not focus when active and the input for addign a highlight is active on form pop up.

As shown in this screenshot:

Screenshot 2024-02-20 at 12 46 10

### Steps to reproduce

1. Go to highlights tab
2. Click on the "post a highlight to show your work" input
3. The "post a highlight" form will pop up
4. Notice that the textarea input will not take an active state even when clicked on while the add a highlight link input is active on load

### Browsers

Chrome

### Additional context (Is this in dev or production?)

dev

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by FatumaA**
## What this PR does?
This PR:
-  adds a focus on hover for the textarea input 
- fixes the permanent error hover in the add a highlight link input

## What type of PR is this?
- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Closes #2715 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/67555014/c092dc6e-f422-4267-9da9-ee0d8b349bce

## Steps to QA
1. Go to highlights tab
2. Click on the "post a highlight to show your work" input
3. The "post a highlight" form will pop up
4. Notice that the text-area input is now focussed when clicked on while the add a highlight link input is no longer focussed on load and is focussed when clicked or when in error

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

N/A

## [optional] What gif best describes this PR or how it makes you feel?
N/A

### CODE REVIEW

File: components/molecules/HighlightInput/highlight-input-form.tsx [hunk 1]
```diff
@@ -642,7 +642,9 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
                 </Tooltip>
                 <TextInput
                   id="highlight-link-input"
-                  className={`text-sm shadow-none h-10 flex-none ${!isHighlightURLValid && "border-red-500"}`}
+                  state={isHighlightURLValid ? "valid" : "invalid"}
+                  className={`text-sm shadow-none h-10 flex-none ${errorMsg && "border-red-500"}`}
+                  // className="text-sm shadow-none h-10 flex-none"
```
Comment (nickytonline) [inline components/molecules/HighlightInput/highlight-input-form.tsx:647]: refactor (blocking): `errorMsg && "border-red-500` will result in `undefined` for a class name if there is no error message.

```suggestion
                  state={isHighlightURLValid ? "valid" : "invalid"}
                  className={`text-sm shadow-none h-10 flex-none ${errorMsg ? "border-red-500": ""}`}
```

You could also use `clsx` which is already in the project as well.

```suggestion
                  state={isHighlightURLValid ? "valid" : "invalid"}
                  className={clsx("text-sm shadow-none h-10 flex-none", errorMsg ? "border-red-500": "")}
```

File: components/molecules/HighlightInput/highlight-input-form.tsx [hunk 2]
```diff
@@ -564,7 +564,7 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
                 </span>
               )}
             </p>
-            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg">
+            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg outline-1 focus-within:outline focus-within:outline-orange-500">
```
Comment (nickytonline) [inline components/molecules/HighlightInput/highlight-input-form.tsx:567]: Textareas in general need default focus states like some other components, but it's fine to do it on an individual level for the moment.

These are the styles we currently use for focus on other form inputs.

```suggestion
            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg focus-within:outline-none focus-within:ring  focus-within:border-orange-500 focus-within:ring-orange-100">
```

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Comment by FatumaA** [comment]
Hi Nicky,
Thank you for the review, I have made the suggested changes.

File: components/molecules/HighlightInput/highlight-input-form.tsx [hunk 3]
```diff
@@ -564,7 +565,7 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
                 </span>
               )}
             </p>
-            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg">
+            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg focus-within:outline-none focus-within:ring  focus-within:border-orange-500 focus-within:ring-orange-100">
```
Comment (nickytonline) [inline components/molecules/HighlightInput/highlight-input-form.tsx:568]: nit: spacing

```suggestion
            <div className="flex flex-col gap-2 p-2 text-sm bg-white border rounded-lg focus-within:outline-none focus-within:ring focus-within:border-orange-500 focus-within:ring-orange-100">
```

**Review by nickytonline — APPROVED** [review]
Thanks @FatumaA!

**Comment by FatumaA** [comment]
> Thanks @FatumaA!

Hi Nicky,
Quick question - you mentioned a uniform type of focus for form inputs across the project, but it appears the input to paste the highlight url in this form has different focus styles than what you suggested.

Should I update the input focus to mirror those styles? I think doing so would ensure visual consistency when interacting with the Post-a-highlight form. 

What are your thoughts?

**Review by NsdHSO — APPROVED** [review]
LGTM

**Review by bdougie — APPROVED** [review]

**Comment by nickytonline** [comment]
There's just a couple conflicts to sort out @FatumaA and then this is ready to go.

CleanShot 2024-04-03 at 07 50 29

**Comment by FatumaA** [comment]
> There's just a couple conflicts to sort out @FatumaA and then this is ready to go.
> 
> CleanShot 2024-04-03 at 07 50 29

please see if its ok, I got into some trouble trying to remove beta history

**Pull request merged**